erpnext_retail/public/js/point_of_sale.js

```diff
@@ -1,28 +1,221 @@
+/* global Payment, __, frappe */
+
 class PointOfSaleNew extends erpnext.pos.PointOfSale{
-	make() {
-		return frappe.run_serially([
-			() => frappe.dom.freeze(),
-			() => {
-				this.prepare_dom();
-				this.prepare_menu();
-				this.set_online_status();
-			},
-			() => this.make_new_invoice(),
-			() => {
-				if(!this.frm.doc.company) {
-					this.setup_company()
-						.then((company) => {
-							this.frm.doc.company = company;
-							this.get_pos_profile();
-						});
-				}
-			},
-			() => {
-				frappe.dom.unfreeze();
-			},
-			() => this.page.set_title(__(this.frm.doc.company + ' Point of Sale'))
-		]);
-	}
+    make() {
+        return frappe.run_serially([
+            () => frappe.dom.freeze(),
+            () => {
+                this.prepare_dom();
+                this.prepare_menu();
+                this.set_online_status();
+            },
+            () => this.make_new_invoice(),
+            () => {
+                if(!this.frm.doc.company) {
+                    this.setup_company()
+                        .then((company) => {
+                            this.frm.doc.company = company;
+                            this.get_pos_profile();
+                        });
+                }
+            },
+            () => {
+                frappe.dom.unfreeze();
+            },
+            () => this.page.set_title(__(this.frm.doc.company + " Point of Sale"))
+        ]);
+    }
 }
 
 erpnext.pos.PointOfSale = PointOfSaleNew
+
+class PaymentNew extends Payment{
+    get_fields() {
+        const me = this;
+
+        let fields = this.frm.doc.payments.map((p) => {
+            return {
+                fieldtype: "Currency",
+                label: __(p.mode_of_payment),
+                options: me.frm.doc.currency,
+                fieldname: p.mode_of_payment,
+                default: p.amount,
+                onchange: () => {
+                    const value = this.dialog.get_value(this.fieldname) || 0;
+                    me.update_payment_value(this.fieldname, value);
+                }
+            };
+        });
+
+        fields = fields.concat([
+            {
+                fieldtype: "Column Break",
+            },
+            {
+                fieldtype: "HTML",
+                fieldname: "numpad"
+            },
+            {
+                fieldtype: "Section Break",
+                depends_on: "eval: this.invoice_frm.doc.loyalty_program"
+            },
+            {
+                fieldtype: "Check",
+                label: "Redeem Loyalty Points",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 tabs but found 16 spaces.](https://app.codacy.com/gh/techlift-tech/erpnext_retail/pullRequest?prid=4838702)